clang/docs/StandardCPlusPlusModules.rst

```diff
@@ -652,6 +652,134 @@ in the future. The expected roadmap for Reduced BMIs as of Clang 19.x is:
    comes, the term BMI will refer to the Reduced BMI and the Full BMI will only
    be meaningful to build systems which elect to support two-phase compilation.
 
+Experimental No Transitive Change
+---------------------------------
+
+Starting with clang19.x, we introduced an experimental feature: the non-transitive
+change for modules, aimed at reducing unnecessary recompilations. For example,
```

Comment:
NIT: Maybe call it "Avoiding unnecessary recompilations"?
It should carry the information people are interested in, but is more grounded in the terms that users are likely to recognize. We could introduce what it does right away:


```
Before Clang 19, a change in BMI of any (transitive) dependency would case the outputs of the BMI to change. Starting with Clang 19, changes to non-direct dependencies should not directly affect the output BMI, unless they affect the results of the compilations.
We expect that there are many more opportunities for this optimization than we currently have realized and would appreaciate feedback about missed optimization opportunities.
... 
```

Having an example after that should clarify this even more.
Does that seems reasonable? (Sorry if I got something wrong, I've probably speculated a bit beyond my undestanding in the comment I wrote)

Reply:
Done. The suggestion looks good. What do you mean for `an example`? There are a following example. Do you mean a example that makes the BMI change but actually it can avoid that?

An example may be:

```
// A.cppm
export module A;
export int a();
```

```
// B.cppm
export module B;
import A;
export inline int b() { return a(); }
```

and we add a unrelated function to A:

```
// A.cppm
export module A;
export int a0();
export int a();
```

Now the users may think the change is unrelated to B and the BMI of B should keep unchanged. But we can't do that since the used DeclID for `a` in module B become to `<1, 1>` instead of `<1, 0>` since the new declaration changes the DeclID. 

Is this the example you want? I didn't post this since I feel it has too many implementation details.

Reply:
Yeah, I meant the example we already have. The current state LG, resolving this thread.